polaris/viz/__init__.py

```diff
@@ -0,0 +1,233 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from matplotlib.collections import PatchCollection
+from matplotlib.colors import LogNorm
+from matplotlib.patches import Polygon
+
+
+def plot_horiz_field(config, ds, ds_mesh, field_name, out_file_name,
+                     title=None, t_index=None, z_index=None,
+                     vmin=None, vmax=None, show_patch_edges=False,
+                     cmap=None, cmap_set_under=None, cmap_scale='linear'):
+
+    """
+    Plot a horizontal field from a planar domain using x,y coordinates at a
+    single time and depth slice.
+
+    Parameters
+    ----------
+    config : polaris.config.PolarisConfigParser
+        Configuration options with parameters used to construct the vertical
+        grid
+
+    ds : xarray.Dataset
+        A data set containing fieldName
+
+    ds_mesh : xarray.Dataset
+        A data set containing mesh variables
+
+    field_name: str
+        The name of the variable to plot, which must be present in ds
+
+    out_file_name: str
+        The path to which the plot image should be written
+
+    title: str, optional
+        The title of the plot
+
+    vmin, vmax : float, optional
+        The minimum and maximum values for the colorbar
+
+    show_patch_edges : boolean, optional
+        If true, patches will be plotted with visible edges
+
+    t_index, z_index: int, optional
+        The indices of 'Time' and 'nVertLevels' axes to select for plotting
+
+    cmap : Colormap or str, optional
+        A color map to plot
+
+    cmap_set_under : str or None, optional
+        A color for low out-of-range values
+
+    cmap_scale : {'log', 'linear'}, optional
+        Whether the colormap is logarithmic or linear
+    """
+    style_path = os.path.realpath(os.path.dirname(__file__))
+    plt.style.use(f'{style_path}/polaris.mplstyle')
+
+    try:
+        os.makedirs(os.path.dirname(out_file_name))
+    except OSError:
+        pass
+
+    if os.path.exists(out_file_name):
+        return
+
+    if title is None:
+        title = field_name
+
+    if 'maxLevelCell' not in ds_mesh:
+        raise ValueError(
+            'maxLevelCell must be added to ds_mesh before plotting.')
+    if field_name not in ds:
+        raise ValueError(
+            f'{field_name} must be present in ds before plotting.')
+
+    field = ds[field_name]
+
+    if 'Time' in field.dims and t_index is None:
+        t_index = 0
+    if t_index is not None:
+        field = field[t_index, :]
+    if 'nVertLevels' in field.dims and z_index is None:
+        z_index = 0
+    if z_index is not None:
+        field = field[:, z_index]
+
+    if 'nCells' in field.dims:
+        ocean_mask = ds_mesh.maxLevelCell - 1 >= 0
+        ocean_patches, ocean_mask = _compute_cell_patches(ds_mesh, ocean_mask)
+    elif 'nEdges' in field.dims:
+        ocean_mask = np.ones_like(field, dtype='bool')
+        ocean_mask = _remove_boundary_edges_from_mask(ds_mesh, ocean_mask)
```

Comment:
FYI, the content of `_remove_boundary_cells_from_mask` is now in `_compute_cell_patches`. I tried to incorporate the content of `_remove_boundary_edges_from_mask` into `_compute_edge_patches` but for an unknown reason it didn't give the same result.

Reply:
I think that's fine anyway. It's a pretty long function.